packages/rs-dpp/src/identity/state_transition/identity_credit_transfer_transition/identity_credit_transfer_transition.rs

```diff
@@ -0,0 +1,253 @@
+        vec![property_names::IDENTITY_ID, property_names::RECIPIENT_ID]
+    }
+
+    fn binary_property_paths() -> Vec<&'static str> {
+        vec![]
+    }
+
+    fn to_object(&self, skip_signature: bool) -> Result<Value, ProtocolError> {
+        let mut value = platform_value::to_value(self)?;
+        if skip_signature {
+            value
+                .remove_values_matching_paths(Self::signature_property_paths())
+                .map_err(ProtocolError::ValueError)?;
+        }
+        Ok(value)
+    }
+
+    fn to_json(&self, skip_signature: bool) -> Result<JsonValue, ProtocolError> {
+        self.to_cleaned_object(skip_signature)
+            .and_then(|value| value.try_into().map_err(ProtocolError::ValueError))
+    }
+
+    fn to_cleaned_object(&self, skip_signature: bool) -> Result<Value, ProtocolError> {
+        let mut value = platform_value::to_value(self)?;
+        if skip_signature {
+            value
+                .remove_values_matching_paths(Self::signature_property_paths())
+                .map_err(ProtocolError::ValueError)?;
+        }
+        Ok(value)
+    }
+}
+
+impl StateTransitionLike for IdentityCreditTransferTransition {
+    fn get_protocol_version(&self) -> u32 {
+        self.protocol_version
+    }
+
+    fn get_type(&self) -> StateTransitionType {
+        self.transition_type
+    }
+
+    fn get_signature(&self) -> &BinaryData {
+        &self.signature
+    }
+
+    fn set_signature(&mut self, signature: BinaryData) {
+        self.signature = signature
+    }
+
+    fn get_modified_data_ids(&self) -> Vec<Identifier> {
+        vec![self.identity_id, self.recipient_id]
+    }
+
+    fn set_signature_bytes(&mut self, signature: Vec<u8>) {
+        self.signature = BinaryData::new(signature)
+    }
+}
+
+impl StateTransitionIdentitySigned for IdentityCreditTransferTransition {
+    /// Get owner ID
+    fn get_owner_id(&self) -> &Identifier {
+        &self.identity_id
+    }
+
+    fn get_signature_public_key_id(&self) -> Option<KeyID> {
+        Some(self.signature_public_key_id)
+    }
+
+    fn set_signature_public_key_id(&mut self, key_id: crate::identity::KeyID) {
+        self.signature_public_key_id = key_id
+    }
+
+    fn get_security_level_requirement(&self) -> Vec<SecurityLevel> {
+        vec![CRITICAL, HIGH, MEDIUM]
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use crate::identity::KeyID;
+    use crate::serialization_traits::{PlatformDeserializable, PlatformSerializable};
+    use crate::state_transition::StateTransitionType;
+    use crate::ProtocolError;
+    use bincode::{config, Decode, Encode};
+    use platform_serialization::{PlatformDeserialize, PlatformSerialize};
+    use platform_value::{BinaryData, Identifier};
+    use rand::Rng;
+    use std::fmt::Debug;
+
+    #[derive(Debug, Clone, Encode, Decode, PlatformDeserialize, PlatformSerialize, PartialEq)]
+    #[platform_error_type(ProtocolError)]
+    pub struct IdentityCreditTransferTransition {
```

Comment:
Why is it there? Why are you overloading the structure with the same name from above?